subsys/net/l2/ieee802154/ieee802154_mgmt.c

```diff
@@ -44,9 +44,7 @@ enum net_verdict ieee802154_handle_beacon(struct net_if *iface,
 		return NET_DROP;
 	}
 
-	if (!mpdu->beacon->sf.association) {
-		return NET_DROP;
-	}
+	ctx->scan_ctx->association_allowed = mpdu->beacon->sf.association;
```

Comment:
While you're on it you could (strictly optional) rename sf.association to sf.<whatever you chose above> for improved readability and consistency.